Bugsnag/KSCrash/Source/KSCrash/Recording/BSG_KSCrashC.c

```diff
@@ -218,7 +218,9 @@ void bsg_kscrash_setThreadTracingEnabled(bool threadTracingEnabled) {
 }
 
 char *bsg_kscrash_captureThreadTrace(int discardDepth, int frameCount, uintptr_t *callstack, const bool recordAllThreads) {
-    BSG_KSCrash_Context *context = crashContext();
+    BSG_KSCrash_Context *globalContext = crashContext();
+    BSG_KSCrash_Context *context = malloc(sizeof(BSG_KSCrash_Context));
```

Comment:
This requires a `free` before returning and a guard for `malloc` returning null (we just return null in that case).

Reply:
Updated this to create a `localContext` variable on the stack instead, as this avoids the need for `malloc` entirely